docs/example_navigation_ros_codegen.md

```diff
@@ -0,0 +1,540 @@
+        twist.linear.y = 0.0;
+        twist.linear.z = 0.0;
+        twist.angular.x = 0.0;
+        twist.angular.y = 0.0;
+        twist.angular.z = ang_vel_cmd;
+        publisher.publish(twist);
+        
+        ROS_INFO("x: %f, y: %f, yaw: %f", current_pos[0], current_pos[1], current_pos[2]);
+        ROS_INFO("Solve time: %f ms. I will send %f %f \n",
+                 (double)status.solve_time_ns / 1000000.0, lin_vel_cmd, ang_vel_cmd);
+
+    }
+    
+}; /* end of class OptimizationEngineManager */
+
+} /* end of namespace open_nmpc_controller */
+
+/**
+ * Main method
+ *
+ * This advertises a new (private) topic to which the optimizer
+ * announces its solution and solution status and details. The
+ * publisher topic is 'open_nmpc_controller/result'.
+ *
+ * It obtains inputs from 'open_nmpc_controller/parameters'.
+ *
+ */
+int main(int argc, char** argv)
+{
+
+    std::string result_topic, params_topic;  /* parameter and result topics */
+    std::string out_twist, in_odometry;
+    double rate; /* rate of node (specified by parameter) */
+
+    open_nmpc_controller::OptimizationEngineManager mng;
+    ros::init(argc, argv, ROS_NODE_MPC_CONTROLLER_NODE_NAME);
+    ros::NodeHandle nh, private_nh("~");
+
+    /* obtain parameters from config/open_params.yaml file */
+    private_nh.param("result_topic", result_topic,
+                     std::string(ROS_NODE_MPC_CONTROLLER_RESULT_TOPIC));
+    private_nh.param("params_topic", params_topic,
+                     std::string(ROS_NODE_MPC_CONTROLLER_PARAMS_TOPIC));
+    private_nh.param("rate", rate,
+                     double(ROS_NODE_MPC_CONTROLLER_RATE));
+    
+    private_nh.param("out_twist_name", out_twist, std::string("/husky_velocity_controller/cmd_vel"));
+    private_nh.param("in_odom_name", in_odometry, std::string("/odometry/filtered"));
+
+    ros::Publisher mpc_pub
+        = private_nh.advertise<open_nmpc_controller::OptimizationResult>(
+            ROS_NODE_MPC_CONTROLLER_RESULT_TOPIC,
+            ROS_NODE_MPC_CONTROLLER_RESULT_TOPIC_QUEUE_SIZE);
+    ros::Subscriber sub
+        = private_nh.subscribe(
+            ROS_NODE_MPC_CONTROLLER_PARAMS_TOPIC,
+            ROS_NODE_MPC_CONTROLLER_PARAMS_TOPIC_QUEUE_SIZE,
+            &open_nmpc_controller::OptimizationEngineManager::mpcReceiveRequestCallback,
+            &mng);
+    ros::Subscriber pos_sub
+        = nh.subscribe(in_odometry,
+                       1,
+                       &open_nmpc_controller::OptimizationEngineManager::odometryCallback,
+                       &mng);
+    ros::Subscriber command_trajectory_subscriber
+        = private_nh.subscribe("command/pose",
+                               1,
+                               &open_nmpc_controller::OptimizationEngineManager::commandPoseCallback,
+                               &mng);
+    ros::Publisher pub_twist_cmd = nh.advertise<geometry_msgs::Twist>(out_twist, 1);
+    ros::Rate loop_rate(ROS_NODE_MPC_CONTROLLER_RATE);
```

Comment:
I think that the line

```c++
ros::Rate loop_rate(ROS_NODE_MPC_CONTROLLER_RATE);
```

should rather be

```c++
ros::Rate loop_rate(rate);
```

and I should in fact chance this in the auto-generated ROS code, otherwise the default rate is always assigned. The same goes for the topic names above.